lib/src/persist/file_adapter.dart

```diff
@@ -1,34 +1,111 @@
-import 'package:casbin/src/model/model.dart';
+// Copyright 2018-2021 The Casbin Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import 'dart:io';
+
+import 'package:csv/csv.dart';
+
+import '../model/model.dart';
 import 'adapter.dart';
 
+void loadPolicyLine(String line, Model model) {
+  if (line.isEmpty || line.trimLeft()[0] == '#') {
+    return;
+  }
+
+  final tokensList = CsvToListConverter().convert(line);
+  if (tokensList.isEmpty || tokensList.first.isEmpty) {
+    return;
+  }
+
+  final tokens = tokensList.first.map((e) => (e as String).trim()).toList();
+
+  final key = tokens.first;
+  final sec = key.substring(0, 1);
+  final item = model.model[sec];
+  if (item == null) {
+    return;
+  }
+
+  final policy = item[key];
+  if (policy == null) {
+    return;
+  }
+
+  policy.policy.add(tokens.sublist(1));
+}
+
+/// FileAdapter is the file adapter for Casbin.
+/// It can load policy from file or save policy to file.
 class FileAdapter implements Adapter {
   final String filePath;
 
   FileAdapter(this.filePath);
 
   @override
-  void addPolicy(String sec, String ptype, List<String> rule) {
-    // TODO: implement addPolicy
+  Future<void> loadPolicy(Model model) async {
+    final lines = await File(filePath).readAsLines();
+    lines.forEach((l) => loadPolicyLine(l, model));
   }
 
   @override
-  void loadPolicy(Model model) {
-    // TODO: implement loadPolicy
+  Future<void> savePolicy(Model model) async {
+    final buffer = StringBuffer();
+
+    final pList = model.model['p'];
+    if (pList == null) {
+      return;
+    }
+    for (var p in pList.entries) {
+      for (var rule in p.value.policy) {
+        buffer.write('${p.key},');
+        buffer.writeln(rule.join(','));
+      }
+    }
+
+    final gList = model.model['g'];
+    if (gList == null) {
+      return;
+    }
+    for (var g in gList.entries) {
+      for (var rule in g.value.policy) {
+        buffer.write('${g.key},');
+        buffer.writeln(rule.join(','));
+      }
+    }
+
+    await File(filePath).writeAsString(buffer.toString().trim());
+    return;
   }
 
   @override
-  void removeFilteredPolicy(
-      String sec, String ptype, int fieldIndex, List<String> fieldValues) {
-    // TODO: implement removeFilteredPolicy
```

Comment:
It may be better to keep the TODO until we implement this, so we can keep track easily.

Reply:
Well, these functions are not available in file adapter implementation.
See also: 
- https://github.com/casbin/casbin/blob/master/persist/file-adapter/adapter.go
- https://github.com/casbin/node-casbin/blob/master/src/persist/fileAdapter.ts